Instrument ActiveStorage analyzers

shouichi opened the pull request:
### Summary

This PR adds instrumentations to ActiveStorage analyzers so that the time of analysis can be tracked.

Close https://github.com/rails/rails/issues/42930.

### Other Information

The following things should be agreed upon before merging.

- Name of the instrumentation event
	- ~Currently `analyzer.active_storage` but should we include analyzer name?~ Changed to `analyze.active_storage` so that it is consistent with others.
- Extra payload to attach (not sure if there any.)
	- Add analyzer name.

Also, we may want to add the new event to the active support instrumentation document.

Thanks.

shouichi committed:
Instrument ActiveStorage analyzers

p8 commented:
This looks great! Can you rebase and add a Changelog entry?

p8 commented:
Could you also add it to the guides? 🙏
https://edgeguides.rubyonrails.org/active_support_instrumentation.html#active-storage

shouichi commented:
Hi, @p8 thanks for the review!

Rebased and made the following changes.

- Changed event name to `analyze.active_storage` so that it is consistent with others (e.g., `transform.active_storage`).
- Added CHANGELOG
- Added the new event to the active support instrumentation guide.

p8 commented on activestorage/CHANGELOG.md:
```diff
@@ -1,3 +1,9 @@
+*   Emit active support events from analyzers
```
```suggestion
*   Emit active support events from analyzers.
```

p8 reviewed (changes requested):
Could you also rebase on `main` to hopefully fix the unrelated spelling warnings?

shouichi commented:
Rebased :+1:

p8 commented:
Thanks! The build failures seem unrelated :/

p8 commented on activestorage/CHANGELOG.md:
```diff
@@ -1,3 +1,9 @@
+*   Emit active support events from analyzers.
```
I think the changelog message is a bit clearer if we mention `instrumentation`:
```suggestion
*   Emit active support instrumentation events from analyzers.
```

shouichi replied:
Thanks for the suggestion! Committed :)

shouichi committed:
Update activestorage/CHANGELOG.md

Co-authored-by: Petrik de Heus <petrik@deheus.net>
